src/flows/context/flow.list.tsx

```diff
@@ -246,32 +257,14 @@ export const FlowListProvider: FC = ({children}) => {
       throw new Error(`${PROJECT_NAME} not found`)
     }
 
-    const data = {
-      name: flow.name,
-      range: flow.range,
-      refresh: flow.refresh,
-      data: flow.data,
-      meta: flow.meta,
-      readOnly: flow.readOnly,
-      results: null,
-    }
-
     setFlows({
       ...flows,
-      [id]: data,
+      [id]: flow,
     })
 
-    const apiFlow = {
-      id,
-      orgID: org.id,
-      data: {
-        id,
-        orgID: org.id,
-        name: flow.name,
-        spec: serialize(data),
-      },
-    }
-    pooledUpdateAPI(apiFlow)
+    const apiFlow = serialize(flow, org.id)
+
+    pooledUpdateAPI({id, ...apiFlow})
```

Comment:
i ended up not having serialize handle the id since a flow being serialized doesn't always have an ID and it was causing issues